### ISSUE TITLE
Timezone switched when filling market

### ISSUE THREAD

**Issue by pgebheim** [AugurProject/augur#1081]
@jfarid27 commented on [Sun Oct 21 2018](https://github.com/AugurProject/augur-ui/issues/2409)

I tried to crate a market and during creation, it was set with timezone UTC -4. After committing to MainNet it somehow switched to UTC -5. I'm not entirely sure why or how to recreate the issue but it seemed weird. Using Chrome 69.0.3497.100 with Metamask on a Linux machine.

---

@adrake33 commented on [Mon Oct 22 2018](https://github.com/AugurProject/augur-ui/issues/2409#issuecomment-432087199)

Internal tracking ticket: https://app.clubhouse.io/augur/story/17667/end-time-is-not-getting-set-properly-when-creating-a-market

---

@phoebemirman commented on [Tue Oct 23 2018](https://github.com/AugurProject/augur-ui/issues/2409#issuecomment-432399258)

this is because of daylight savings time, utc offset differs by an hour depending on whether it is between march 11th and nov 4th.

---

@0xpunk commented on [Thu Feb 14 2019](https://github.com/AugurProject/augur-ui/issues/2409#issuecomment-463920999)

Why was this issue closed? Yes it is because of ~~timezone~~ daylight saving time but it is still a big problem that the UI shows a different timezone when entering the expiration, than the one that will actually be used!

At no point is this visible until after the market is created and it is too late to change it. I just got bit by this issue myself and now have no idea what to do, except kiss my staked REP and ETH goodbye...

When you go to create a market, the expiration field is labeled "EXPIRATION TIME (UTC -1)". Obviously this very strongly implies that you are entering a UTC -1 time. Only after you submit the tx to actually create the market can you see that it actually took your time and used UTC -2, so the market actually expires one hour earlier than intended!

If the app is going to adjust my time for DST why wouldn't that be done BEFORE submitting the market, when you still have a chance to add it correctly?? Just update the timezone in the label when user sets the date... or better yet just use UTC everywhere and avoid this nonsense altogether

Any advice on what to do with my broken market? It expires one hour too early now, can I assume that it should resolve Invalid and report it so? Now I am worried that someone with more REP can just say "nah, one hour is close enough" and dispute it...

Sorry for the long rant but needless to say this is a bit stressful! I feel like I just burned almost 9 REP and to add insult to injury, the validity bond too... FML LOL

---

@pgebheim commented on [Wed Feb 20 2019](https://github.com/AugurProject/augur-ui/issues/2409#issuecomment-465490838)

We have some other tickets related to timezone handling re-design already scheduled, I'm gonna transfer this issue into the monorepo and add it to that epic.

Thanks for the report @0xpunk -- will add a link to the new issue...

**Comment by 0xpunk** [comment]
Thanks @pgebheim. I think the solution is pretty simple: do the same adjustment for DST you are doing now (it is correct after all), but do it on the screen where the user sets the expiration date and time, instead of after submitting, and update the label accordingly. IOW: the expiration form presently shows the local time zone "right now," it should actually show the local timezone *on* the expiration date, and update the label accordingly (so the label has to be updated when the date selection changes).

Related to this, on the final confirmation screen the time is shown with no timezone at all. Obviously I think the correct DST-adjusted local timezone should be shown instead. But also, I think the corresponding UTC time should be displayed as well. The reason is that often when creating a market, you want to target a UTC time in the first place (UTC midnight is extremely common), so it would be better to see a clear confirmation of that. It just makes perfect sense IMO, after all you are committing real funds to an immutable contract, it should let you see exactly what you're committing to.

**Comment by pgebheim** [comment]
@bthaile @cillianhunter0 -- Can you decide how this will apply NOW and also in the full redesign and split up tickets / do what needs to be done?

Anything for now should stay in v1.11 release, anything that goes into the new trading designs v1.12

**Comment by joeykrug** [comment]
this is fixed now
